### ISSUE TITLE
请问使用zero2/zero3导致max_steps相差八倍的原因是什么？

### ISSUE THREAD

**Issue by HT-NEKO** [modelscope/ms-swift#4616]
我的脚本：
```
export WANDB_API_KEY=xxx
export WANDB_MODE=dryrun
export WANDB_PROJECT=ms-swift
export WANDB_RUN_NAME=Qwen3-1.7B

NPROC_PER_NODE=8
CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7
MASTER_PORT=56789
PYTORCH_CUDA_ALLOC_CONF='expandable_segments:True'
export VLLM_USE_V1=1
swift rlhf \
    --rlhf_type grpo \
    --model /primus_source_checkpoint/Qwen3-1_7B \
    --model_type "qwen3" \
    --external_plugins /root/code/examples/train/grpo/plugin/plugin.py \
    --reward_funcs external_countdown format \
    --use_vllm true \
    --vllm_mode colocate \
    --vllm_max_model_len 2048 \
    --sleep_level 1 \
    --offload_optimizer true \
    --offload_model true \
    --gc_collect_after_offload true \
    --vllm_gpu_memory_utilization 0.5 \
    --train_type lora \
    --torch_dtype bfloat16 \
    --dataset 'zouxuhong/Countdown-Tasks-3to4#50000' \
    --max_length 2048 \
    --max_completion_length 1024 \
    --num_train_epochs 1 \
    --per_device_train_batch_size 8 \
    --per_device_eval_batch_size 8 \
    --learning_rate 5e-7 \
    --gradient_accumulation_steps 8 \
    --eval_steps 100 \
    --save_steps 100 \
    --save_total_limit 3 \
    --logging_steps 1 \
    --output_dir output/GRPO_COUNTDOWN \
    --warmup_ratio 0.01 \
    --dataloader_num_workers 8 \
    --num_generations 8 \
    --temperature 1.0 \
    --system 'You are a helpful assistant. You first thinks about the reasoning process in the mind and then provides the user with the answer.' \
    --deepspeed zero2 \
    --log_completions true \
    --report_to wandb \
    --beta 0.001 \
    --num_iterations 1 \
    --sequence_parallel_size 8 \
    --tensor_parallel_size 8
```
如果使用```--deepspeed zero2```，max_steps=97：
{"loss": 6.7e-07, "grad_norm": 0.04683236, "learning_rate": 4.6e-07, "memory(GiB)": 42.21, "train_speed(iter/s)": 0.010316, "completions/mean_length": 750.1328125, "completions/min_length": 238.0, "completions/max_length": 1025.0, "completions/clipped_ratio": 0.45703125, "rewards/CountdownORM/mean": 0.54101562, "rewards/CountdownORM/std": 0.49880221, "rewards/Format/mean": 0.52929688, "rewards/Format/std": 0.49962911, "reward": 1.0703125, "reward_std": 0.34645408, "kl": 0.00067735, "clip_ratio/low_mean": 0.0, "clip_ratio/low_min": 0.0, "clip_ratio/high_mean": 0.0, "clip_ratio/high_max": 0.0, "clip_ratio/region_mean": 0.0, "epoch": 0.18619686, **"global_step/max_steps": "18/97"**, "percentage": "18.56%", "elapsed_time": "26m 35s", "remaining_time": "1h 56m 43s"}

如果使用```--deepspeed zero3```，max_steps=774：
{"loss": -0.0, "grad_norm": 0.0, "learning_rate": 5e-07, "memory(GiB)": 31.54, "train_speed(iter/s)": 0.024826, "completions/mean_length": 767.3125, "completions/min_length": 218.0, "completions/max_length": 1025.0, "completions/clipped_ratio": 0.4453125, "rewards/CountdownORM/mean": 0.546875, "rewards/CountdownORM/std": 0.49828473, "rewards/Format/mean": 0.54101562, "rewards/Format/std": 0.49880221, "reward": 1.08789062, "reward_std": 0.29182103, "kl": 0.0, "clip_ratio/low_mean": 0.0, "clip_ratio/low_min": 0.0, "clip_ratio/high_mean": 0.0, "clip_ratio/high_max": 0.0, "clip_ratio/region_mean": 0.0, "epoch": 0.02327461, **"global_step/max_steps": "18/774"**, "percentage": "2.33%", "elapsed_time": "9m 14s", "remaining_time": "6h 28m 4s"}

**Comment by Jintao-Huang** [comment]
please use main branch or ms-swift3.5.1

**Comment by HT-NEKO** [comment]
> please use main branch or ms-swift3.5.1

感谢！！
